### DISCUSSION TITLE
Google TagManager snippet is affecting total block time in pageinsights

### DISCUSSION THREAD

**Discussion by angrejkumar** [vercel/next.js#59367] [Help]
### Summary

In my NextJS application I have added google tag manager embed code like this in a separate component and then I call this component in _app.tsx, but when I add this my https://pagespeed.web.dev/ Total Blocking Time increased. I think google tag manager is loading asynchronously, but not me for me I think. Not sure what wrong I'm doing.

Can please some help? I'm using NextJS 13.3

### Additional information

```js
"use client"

import { GTM_ID, pageview } from "@/lib/gtm"
import { usePathname, useSearchParams } from "next/navigation"
import Script from "next/script"
import { useEffect } from "react"

export default function Analytics() {
  const pathname = usePathname()
  const searchParams = useSearchParams()

  useEffect(() => {
    if (pathname) {
      pageview(pathname)
    }
  }, [pathname, searchParams])

  if (process.env.NODE_ENV !== "production") {
    return null
  }

  return (
    <>
      <noscript>
        <iframe
          src={`https://www.googletagmanager.com/ns.html?id=${GTM_ID}`}
          height="0"
          width="0"
          style={{ display: "none", visibility: "hidden" }}
        />
      </noscript>
      <Script
        id="gtm-script"
        strategy="afterInteractive"
        dangerouslySetInnerHTML={{
          __html: `
    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
    })(window,document,'script','dataLayer', '${GTM_ID}');
  `,
        }}
      />
    </>
  )
}
```

**Comment by srhise** [comment]
This is also occuring for me. However, I am using the `<GoogleTagManager/>` component from @next/third-parties. The scripts loaded from GTM are hoisted into the <head>, marked for preload, and block the main thread for 600ms.

**Reply by srhise** [reply]
<img width="843" alt="image" src="https://github.com/vercel/next.js/assets/914113/bd7781d7-9cdb-4c8b-b4ff-3ac56811928b">

**Comment by hexionaut** [comment]
Also noticed that with `@next/third-parties` would be good to have an option to defer.
